sql/auth/sql_user.cc

```diff
@@ -2978,7 +2985,15 @@ bool mysql_create_user(THD *thd, List<LEX_USER> &list, bool if_not_exists,
     User_params user_params(&extra_users);
     result = log_and_commit_acl_ddl(thd, transactional_tables, &extra_users,
                                     &user_params);
+#ifndef WITH_WSREP
+    // clang-format off
+    /* We already started TOI. Below we will acquire ACL lock again.
```

Comment:
Does this problem exist with only CREATE TABLE? i.e, are there any other ACL statements which release the ACL lock temporarily and reacquire it later?

Reply:
I reviewed all cases where `acl_cache_lock` is used together with `start_toi_after_open_grant_tables` and found that besides `CREATE USER`, there is `ALTER USER` which suffers the same problem (exiting and re-entering critical section). Fixed.